site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+</div>
+
+---
+
+
+## Interacting with controls
+
+### Active and disabled
+Form controls are either *active* or *disabled* depending on for example selections made in previous controls in the form, or user permissions. 
+
+- **Active form controls can be focused and given a value.** Form components are active by default
+- **Disabled form controls cannot be clicked or focused with keyboard.**
+    - The disabled attribute is be set for a form control to keep a user from using it until some other condition has been met (for example a text input can be disabled before a checkbox is selected).
+
+<Image src="../../static/patterns/form/form-states-active-disabled@2x.png" alt="Active and disabled form controls" style="max-width:376px;" viewable /> 
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li>Disabled form controls can also be hidden from view, but in most situations all fields, even the disabled ones should be shown to the user.
+            <ul>
+                <li>Showing all fields makes easier or the user to understand all the possible actions in the form.</li>
+                <li>Previously hidden fields suddenly appearing to a form can confuse the user.</li>
+                <li>As a rule of thumb, the field should only be hidden, if the user does not have permission to change it’s value.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+---
+
+### Mouse over
+When the mouse cursor is over an active form control, a visual hover cue indicates the user that the form field can be interacted with.
+
+<Image src="../../static/patterns/form/form-states-hover@2x.png" alt="Hover indicator" style="max-width:376px;" viewable /> 
+
+---
+
+### Focus
+**An active form control receives focus** when the user either clicks the form control with mouse or navigates into it with keyboard. When the form control is focused, the user can fill or select values to it.
+
+- When a form component receives focus:
+    - A visual focus indicator tells a seeing user which form element is currently being interacted with.
+    - Screen reader reads the label and assistive text
+- Also the interactive elements inside a form control, for example action buttons and tags, can be focused independently when interacting with the main control.
+
+See [Navigating with keyboard section below](/#navigating-with-keyboard) for more information on helping the user navigate forms with keyboards.
+
+<Image src="../../static/patterns/form/form-states-focus@2x.png" alt="Focus indicator" style="max-width:376px;" viewable /> 
+
+---
+
+### Empty/unselected and filled/selected
```

Comment:
> Empty/unselected and filled/selected

Myös tämä kappale tuntuu turhan pieniltä yksityiskohdilta. Tuossa "do/don't" -osiossa on kuitenkin todella hyvää tavaraa.

Reply:
Jätin tämän, muotoilin johtolausetta niin, että se korostaa tämän kohdan pointtia, eli noita do ja don't ohjeita:
"Allowing the user enter a value either by input or selection is the form controls basic function. Make it as easy and straightforward as possible for the user."